Dockerfile.java

```diff
@@ -0,0 +1,46 @@
+# radanalytics-java-spark
+FROM fabric8/s2i-java
+
+MAINTAINER Trevor McKay tmckay@redhat.com
+ 
+ENV RADANALYTICS_JAVA_SPARK 1.0
+
+LABEL io.k8s.description="Platform for building a radanalytics java spark app" \
+      io.k8s.display-name="radanalytics java_spark" \
+      io.openshift.expose-services="8080:http" \
+      io.openshift.s2i.scripts-url="image:///usr/local/s2i" \
+      io.openshift.tags="builder,radanalytics,java_spark"
+
+USER root
+
+RUN yum install -y epel-release tar java && \
+    yum clean all
+
+RUN cd /opt && \
+    curl https://dist.apache.org/repos/dist/release/spark/spark-2.0.1/spark-2.0.1-bin-hadoop2.7.tgz | \
+        tar -zx && \
+    ln -s spark-2.0.1-bin-hadoop2.7 spark
+
+RUN yum install -y golang make && \
+    yum install -y nss_wrapper && \
+    yum clean all
+
+ENV GOPATH /go
+ADD . /go/src/github.com/radanalyticsio/oshinko-s2i
+
+ENV APP_ROOT /opt/app-root
+
+RUN mkdir -p $APP_ROOT/src && mkdir $APP_ROOT/etc && \
```

Comment:
So here we have to create the $APP_ROOT/etc dir ourselves and carry a copy of the generate_container_user script from the python s2i image since our shared start.sh is expecting it $APP_ROOT/etc. It was either that or pull a copy of the old common.sh from git history and have a separate start.sh .....